### PR TITLE
[Shape Fns] Fix handling of empty dim list in sum_mean_dim shape fn

### PR DESCRIPTION
**Pull request by ramiro050**
The current implementation of the `sum_mean_dim` shape function
takes `dim=[]` and `dim=None` to mean "no reduction". However, in the
ops `torch.sum` and `torch.mean`, both `dim=[]` and `dim=None` are
equivalent to "reduce along all dimensions". This commit fixes the
handling of `dim` in the `sum_mean_dim` shape function.

### CODE REVIEW

**Review by Gamrix — APPROVED** [review]
Looks good, I looked into why the opinfo tests weren't catching this bug, and it turned out that the dim is empty case is just being skipped.

torch/testing/_internal/common_methods_invocations.py
```
            # FIXME: mean reduces all dimensions when dim=[]
            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_empty'),
            DecorateInfo(unittest.skip("Skipped!"), 'TestReductions', 'test_dim_empty_keepdim'),
 ```

**Comment by ramiro050** [comment]
Thanks for reviewing my PR @Gamrix ! What is the merging protocol here?

**Comment by Gamrix** [comment]
Sorry, somehow I thought you were FB-internal and had access to kick off the merge process. Thanks for the flag, I have started the process now.
